round 전체 완성

JaehoonJung0430 opened the pull request:
round 부분 전체 완성(Round, RoundThumbnail 일부 추가, RoundThumbnailDTO, RoundController, RoundRepository, RoundThumbnailRepository, RoundService 추가)

JungJaehoon0430 committed:
round 전체 완성

round 부분 전체 완성(Round, RoundThumbnail 일부 추가, RoundThumbnailDTO, RoundController, RoundRepository, RoundThumbnailRepository, RoundService 추가)

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/controller/RoundController.java:
```diff
@@ -0,0 +1,70 @@
+  @Operation(summary = "회차 생성", description = "새로운 회차를 생성합니다.")
+  public ResponseEntity<RoundDTO> createRound(
+          @Parameter(description = "스터디 ID", required = true) @PathVariable Long studyId,
+          @RequestParam String goal,
```
묶어서 requestDTO 하나 만드는게 좋아보여요

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/controller/RoundController.java:
```diff
@@ -0,0 +1,70 @@
+  public ResponseEntity<RoundDTO> updateRound(
+          @Parameter(description = "스터디 ID", required = true) @PathVariable Long studyId,
+          @Parameter(description = "회차 ID", required = true) @PathVariable Long roundId,
+          @RequestParam String goal,
```
여기도 묶어서 DTO로 바꾸는게 좋아보여요

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/domain/Round.java:
```diff
@@ -10,11 +10,17 @@
 
 @Entity
+@Getter
+@NoArgsConstructor
```
다른 분들이랑 통일해서 noarg 는 protected로,
allarg 는 private으로 만들고,
builder annotation 은 클래스 위로 옮기는게 어떨까요

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/domain/RoundThumbnail.java:
```diff
@@ -9,8 +9,13 @@
 
 @Entity
+@Getter
+@NoArgsConstructor
```
여기도 위처럼 noarg, allarg, builder 같은 이야기입니다

cmj7271 commented on src/main/java/com/gdgoc/study_group/round/service/RoundService.java:
```diff
@@ -0,0 +1,139 @@
+
+@Service
+@RequiredArgsConstructor
+@Transactional
```
readOnly 로 바꿔주세요
자세한 내용은 디스코드에 적혀있어요

JungJaehoon0430 committed:
회차 생성, 수정 요청 DTO 수정, 일부 변화

CreateRoundRequest, UpdateRoundRequest 생성
컨트롤러, 도메인, dto, service 일부 수정

@Operation(summary = "회차 생성", description = "새로운 회차를 생성합니다.")
  public ResponseEntity<RoundDTO> createRound(
          @Parameter(description = "스터디 ID", required = true) @PathVariable Long studyId,
          @RequestParam String goal,
Member
묶어서 requestDTO 하나 만드는게 좋아보여요

src/main/java/com/gdgoc/study_group/round/controller/RoundController.java
  public ResponseEntity<RoundDTO> updateRound(
          @Parameter(description = "스터디 ID", required = true) @PathVariable Long studyId,
          @Parameter(description = "회차 ID", required = true) @PathVariable Long roundId,
          @RequestParam String goal,
Member
여기도 묶어서 DTO로 바꾸는게 좋아보여요

@Entity
@Getter
@NoArgsConstructor
Member
다른 분들이랑 통일해서 noarg 는 protected로,
allarg 는 private으로 만들고,
builder annotation 은 클래스 위로 옮기는게 어떨까요

@Entity
@Getter
@NoArgsConstructor
Member
여기도 위처럼 noarg, allarg, builder 같은 이야기입니다

@Service
@RequiredArgsConstructor
@Transactional
Member@cmj7271
readOnly 로 바꿔주세요
자세한 내용은 디스코드에 적혀있어요

cmj7271 commented:
추후에 DTO 명칭 다듬기로 하고 일단 머지

cmj7271 reviewed (approved):
LGTM
